egs/wsj/s5/utils/lang/make_lexicon_fst.py

```diff
@@ -0,0 +1,407 @@
+                        help="""Probability of silence between words (including at the
+                        beginning and end of word sequences).  Must be in the range [0.0, 1.0].
+                        This refers to the optional silence inserted by the lexicon; see
+                        the --silphone option.""")
+    parser.add_argument('--sil-disambig', dest='sil_disambig', type=str,
+                        help="""Disambiguation symbol to disambiguate silence, e.g. #5.
+                        Will only be supplied if you are creating the version of L.fst
+                        with disambiguation symbols, intended for use with cyclic G.fst.
+                        This symbol was introduced to fix a rather obscure source of
+                        nondeterminism of CLG.fst, that has to do with reordering of
+                        disambiguation symbols and phone symbols.""")
+    parser.add_argument('--left-context-phones', dest='left_context_phones', type=str,
+                        help="""Only relevant if --nonterminals is also supplied; this relates
+                        to grammar decoding (see http://kaldi-asr.org/doc/grammar.html or
+                        src/doc/grammar.dox).  Format is a list of left-context phones,
+                        in text form, one per line.  E.g. data/lang/phones/left_context_phones.txt""")
+    parser.add_argument('--nonterminals', type=str,
+                        help="""If supplied, --left-context-phones must also be supplied.
+                        List of user-defined nonterminal symbols such as #nonterm:contact_list,
+                        one per line.  E.g. data/local/dict/nonterminals.txt.""")
+    parser.add_argument('lexiconp', type=str,
+                        help="""Filename of lexicon with pronunciation probabilities
+                        (normally lexiconp.txt), with lines of the form 'word prob p1 p2...',
+                        e.g. 'a   1.0    ay'""")
+    args = parser.parse_args()
+    return args
+
+
+def read_lexiconp(filename):
+    """Reads the lexiconp.txt file in 'filename', with lines like 'word pron p1 p2 ...'.
+    Returns a list of tuples (word, pron_prob, pron), where 'word' is a string,
+   'pron_prob', a float, is the pronunciation probability (which must be >0.0
+    and would normally be <=1.0),  and 'pron' is a list of strings representing phones.
+    An element in the returned list might be ('hello', 1.0, ['h', 'eh', 'l', 'ow']).
+    """
+
+    ans = []
+    found_empty_prons = False
+    found_large_pronprobs = False
+    # See the comment near the top of this file, RE why we use latin-1.
+    with open(filename, 'r', encoding='latin-1') as f:
+        for line in f:
+            a = line.split()
+            if len(a) < 2:
+                print("{0}: error: found bad line '{1}' in lexicon file {1} ".format(
+                    sys.argv[0], line.strip(), filename), file=sys.stderr)
```

Comment:
The "{1}" appears twice in string formatting. The second "{1}" should be "{2}". It also happens in later parts.